### PR TITLE
Hydra changes to lightning-lite

### LINKED ISSUE #14799: Bring Hydra improvements to Lite
## Proposed refactor

Apply/share changes in #11617 with the Lite implementation.

### Motivation

Since both are using the same launcher, the improvements should be integrated for both.

### Pitch

Move the  `_hydra_subprocess_cmd` functions to `lightning_lite.strategies.lauchers.subprocess`, and share the implementation in PL. 

### Additional context

#11617 was developed in parallel to standalone Lite efforts, so we didn't have time to adjust. 

______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

- [**Metrics**](https://github.com/Lightning-AI/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Lite**](https://pytorch-lightning.readthedocs.io/en/latest/starter/lightning_lite.html): enables pure PyTorch users to scale their existing code on any kind of device while retaining full control over their own loops and optimization logic.

- [**Flash**](https://github.com/Lightning-AI/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, fine-tuning, and solving problems with deep learning.

- [**Bolts**](https://github.com/Lightning-AI/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks, and more for research and production with PyTorch Lightning and PyTorch.

- [**Lightning Transformers**](https://github.com/Lightning-AI/lightning-transformers): Flexible interface for high-performance research using SOTA Transformers leveraging PyTorch Lightning, Transformers, and Hydra.

### PR DESCRIPTION
**Pull request by Atharva-Phatak**
## What does this PR do?

This PR adds the same changes #11617 to lightning lite launcher. This PR addresses the issue #14799 .

Fixes #14799 

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [ ] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

File: src/pytorch_lightning/strategies/launchers/subprocess_script.py [hunk 1]
```diff
@@ -155,7 +155,7 @@ def _hydra_subprocess_cmd(local_rank: int) -> Sequence[str]:
     else:
         command = [sys.executable, "-m", __main__.__spec__.name]
 
-    # extract the hydra configu
```
Comment (awaelchli) [inline src/pytorch_lightning/strategies/launchers/subprocess_script.py:158]: You can now delete this function, and the one above, and then import it from 

`lightning_lite.strategies.launchers.subprocess_script`

**Review by awaelchli — COMMENTED** [review]
Great! One minor comment. 

And I will take a look at the tests later on.

File: src/pytorch_lightning/strategies/launchers/subprocess_script.py (hunk 1 above)
Comment (Atharva-Phatak) [reply src/pytorch_lightning/strategies/launchers/subprocess_script.py:158]: Oh, so you want that the _hydra_subprocess function should be the default for lightning_lite and be imported inside the PL subprocess script? Right ? 

I can take a look at tests though not sure how to go about it.

Comment (Atharva-Phatak) [reply src/pytorch_lightning/strategies/launchers/subprocess_script.py:158]: @awaelchli I have made the changes let me know if you need anything from my end.

**Review by carmocca — APPROVED** [review]

File: src/lightning_lite/strategies/launchers/subprocess_script.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 from lightning_lite.plugins.environments.cluster_environment import ClusterEnvironment
 from lightning_lite.strategies.launchers.base import _Launcher
 
-_HYDRA_AVAILABLE = RequirementCache("hydra")
+_HYDRA_AVAILABLE = RequirementCache("hydra-core")
```
Comment (awaelchli) [inline src/lightning_lite/strategies/launchers/subprocess_script.py:27]: This was a bug, we forgot to update it. It is also like this in the PL version. I added a test. 
With this, the PR is complete I think.

**Review by awaelchli — APPROVED** [review]

File: src/pytorch_lightning/strategies/launchers/subprocess_script.py (hunk 1 above)
Comment (awaelchli) [reply src/pytorch_lightning/strategies/launchers/subprocess_script.py:158]: Yeah, that's great! exactly what we needed. And I added a test.
